plugins/dns-discovery/src/main/java/io/crate/discovery/SrvUnicastHostsProvider.java

```diff
@@ -117,7 +118,9 @@ InetSocketAddress parseResolverAddress(Settings settings) {
 
     private DnsNameResolver buildResolver(Settings settings) {
         DnsNameResolverBuilder resolverBuilder = new DnsNameResolverBuilder(eventLoopGroup.next());
-        resolverBuilder.channelType(NioDatagramChannel.class);
+        resolverBuilder.datagramChannelType(NioDatagramChannel.class);
+        resolverBuilder.socketChannelType(NioSocketChannel.class, true);
+        resolverBuilder.queryTimeoutMillis(Math.min(Math.abs(resolveTimeout.millis() / 2), 100));
```

Comment:
I've set the timeout lower than the configured timeout as netty will always try to connect over UDP first and fallsback to TCP once the UDP request times out. To still hit the TCP fallback, the initial timeout must be lower than the overall timeout set to wait for any DNS resolving.
But it's unclear to me, what lower value to use. Any idea?

Reply:
There's probably no "correct" heuristic, so anything like half/half should be fine. Maybe increase the total to 10 seconds from 5?

Question is if we also add a dedicated setting - but I'd probably only do that if we encounter a scenario where it is required to ensure we can also document in which case it should be changed.

Afaik there's no harm if the timeout for this is a bit longer? 

Alternative - could we do both in parallel and have them race - e.g. whatever is first wins?

Reply:
> Forcing TCP only based on an exposed setting would be also an option but isn't supported by netty atm, see https://github.com/netty/netty/issues/11130.

Could we emulate the TCP only with a really short UDP timeout of a couple ms?
Then we could use two resolvers - one with the regular normal timeout, one with the short UDP timeout, and use the result of whichever was faster.

Reply:
Good idea, I'll check, thx!

Reply:
I pushed a draft implementing this. But I think this approach is not working well as we can only set a overall `queryTimeout`. Such, when setting the timeout low on the TCP resolver in order to fallback to TCP fast, the same setting will also be used for the follow up TCP request.

Reply:
Then I'd say let's go with the half/half logic for the hotfix, maybe increasing the default *2, so that the udp timeout overall stays the same, but we additionally have the TCP attempts afterwards with a longer total timeout.

If this ends up being a problem in practice, we could consider contributing to the netty issue.